src/components/ImageEditor.vue

```diff
@@ -142,22 +140,39 @@ export default {
 		},
 
 		async onSave(imageData) {
-			const filename = join(dirname(this.filename), imageData.fullName)
-			logger.debug('Saving image...', { src: this.src, filename })
+			const { origin, pathname } = new URL(this.src)
+			const putUrl = origin + join(dirname(pathname), imageData.fullName)
+			logger.debug('Saving image...', { putUrl, src: this.src, fullName: imageData.fullName })
+
 			try {
-				const b64string = imageData.imageBase64.split(';base64,').pop()
-				const buff = Buffer.from(b64string, 'base64')
-				await client.putFileContents(filename, buff, {
-					// @see https://github.com/perry-mitchell/webdav-client#putfilecontents
-					// https://github.com/perry-mitchell/webdav-client/issues/150
-					contentLength: false,
-				})
+				const file = this.dataURLtoFile(imageData.imageBase64, imageData.fullName)
+
+				await axios.put(putUrl, file)
+
 				logger.info('Edited image saved!')
+				if (putUrl !== this.src) {
+					emit('files:file:created', putUrl)
```

Comment:
If we emit `putUrl` only, then the recipient won't have the file id, making it hard to know which file we are talking about

Reply:
Yeah, I also was thinking about this.
It's kind of hard to actually get all the data we want about this.

I was wondering if we could tune our dav server to return a file Info.

Reply:
What I don't like is having different return object for similar events. Not very consistent.

Though I could return the FileID from the put HEADER

Reply:
Done!